src/package.coffee

```diff
@@ -43,9 +43,9 @@ class Package
           @normalizeMetadata(metadata)
         catch error
           throw error unless ignoreErrors
-
-    metadata ?= {}
-    metadata.name = packageName
+      else
+        metadata ?= {}
```

Comment:
This still needs to run in all cases since ` CSON.readFileSync(metadataPath)` might return null or if an error is thrown and `ignoreErrors` is `true`